kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -156,6 +156,111 @@ static int findMaxColumnId(StructType schema) {
     return maxColumnId;
   }
 
+  static void validateColumnIds(StructType currentSchema, StructType updatedSchema) {
+    Map<Long, String> currentFieldIdToPhysicalName = new HashMap<>();
+    for (StructField field : currentSchema.fields()) {
+      validateColumnIds(field, currentFieldIdToPhysicalName);
+    }
+
+    Map<Long, String> updatedFieldIdToPhysicalName = new HashMap<>();
+    for (StructField field : updatedSchema.fields()) {
+      validateColumnIds(field, updatedFieldIdToPhysicalName);
+    }
+
+    Set<String> dedupedPhysicalNames = new HashSet<>(updatedFieldIdToPhysicalName.values());
+    if (dedupedPhysicalNames.size() != updatedFieldIdToPhysicalName.size()) {
+      throw new IllegalArgumentException("Assigned physical names must be unique");
+    }
+
+    for (Map.Entry<Long, String> field : updatedFieldIdToPhysicalName.entrySet()) {
+      String existingPhysicalName = currentFieldIdToPhysicalName.get(field.getKey());
+      // Found an existing field, verify the physical name is preserved
+      if (existingPhysicalName != null && !existingPhysicalName.equals(field.getValue())) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Existing field with id %s in current schema has physical name %s which is different from %s",
+                field.getKey(), existingPhysicalName, field.getValue()));
+      }
+    }
+  }
+
+  private static void validateColumnIds(
```

Comment:
comment describing what this function does

Reply:
Done!